subworkflows/local/input_check_screening.nf

```diff
@@ -0,0 +1,45 @@
+//
+// Check input samplesheet and get read channels
+//
+
+include { SAMPLESHEET_CHECK_SCREENING } from '../../modules/local/samplesheet_check_screening'
+
+workflow INPUT_CHECK_SCREENING {
+    take:
+    samplesheet // file: /path/to/samplesheet.csv
+
+    main:
+    SAMPLESHEET_CHECK_SCREENING ( samplesheet )
+        .csv
+        .splitCsv ( header:true, sep:',' )
+        .map { create_fastq_channel(it) }
+        .set { reads }
+
+    emit:
+    reads                                     // channel: [ val(meta), [ reads ] ]
+    versions = SAMPLESHEET_CHECK_SCREENING.out.versions // channel: [ versions.yml ]
+}
+
+// Function to get list of [ meta, [ fastq_1, fastq_2 ] ]
+def create_fastq_channel(LinkedHashMap row) {
+    // create meta map
+    def meta = [:]
+    meta.id         = row.sample
+    meta.single_end = row.single_end.toBoolean()
```

Comment:
Is `single_end` being determined automatically if a second fastq file is provided or is this a relict from the template? From the samplesheet and docs, this is not available/needed.